core/src/main/scala/org/apache/spark/sql/rapids/tool/qualification/QualificationAppInfo.scala

```diff
@@ -795,6 +795,10 @@ class QualificationAppInfo(
     val sqlPlanInfoGraphEntry = SqlPlanInfoGraphBuffer.createEntry(sqlID, planInfo)
     checkMetadataForReadSchema(sqlPlanInfoGraphEntry)
     for (node <- sqlPlanInfoGraphEntry.sparkPlanGraph.allNodes) {
+      if (node.name.startsWith("Photon")) {
```

Comment:
I thought that we could detect the Photon in an earlier phase as we process the eventlogs.
For example, `doSparkListenerSQLExecutionStart()` could check for certain conditions to disqualify the entire app.
The pros of having the code in  `doSparkListenerSQLExecutionStart()`:

- For Qualification tool, the bottleneck of performance is actually in reading and processing the evntlogs. Therefore, if we fail during that stage, then we are saving significant amount of useless processing and computation.
- If we need to propagate that logic to Profiling, it becomes much easier as we need only to move the implementation to the `EvenProcessorBase.doSparkListenerSQLExecutionStart` instead of `QualificationEventProcessor.doSparkListenerSQLExecutionStart`. Qualification if we need